### PR TITLE
Blog Post: Using Kubernetes Configuration Provider to load data from Secrets and Config Maps

### PR DESCRIPTION
**Pull request by scholzj**
This PR adds a new Blog Post about _Using Kubernetes Configuration Provider to load data from Secrets and Config Maps_.

### CODE REVIEW

File: _posts/2021-07-20-using-kubernetes-config-provider-to-load-data-from-secrets-and-config-maps.md [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+
+This Role has to be created in the namespace where the Secret or Config Map we want to use exists.
+In the example above, this Role will give access only to a single Secret named `db-credentials` in namespace `database`.
+And the only operation it allows is `get` to read the Secret.
+This way, the configuration provider with this Role will not be able to modify the secret or read any other secrets.
+You can of course customize the Role.
+You can specify multiple Secrets or add Config Maps as well.
+It all depends on your requirements.
+
+Next, we will ned to create a Role Binding to assign this Role to the Kafka Connect Pods:
```
Comment (Frawless) [inline _posts/2021-07-20-using-kubernetes-config-provider-to-load-data-from-secrets-and-config-maps.md:164]:
```suggestion
Next, we will need to create a Role Binding to assign this Role to the Kafka Connect Pods:
```

**Review by Frawless — APPROVED** [review]
Just two nits, overall it looks good to me with a lot of very useful info 👍

File: _posts/2021-07-22-using-kubernetes-config-provider-to-load-data-from-secrets-and-config-maps.md [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+<!--more-->
+
+Usually, when you need to use data from a Config Map or Secret in your Pod, you will either mount it as volume or map it to an environment variable.
+Both methods are configured in the `spec` section or the Pod resource or in the `spec.template.spec` section when using higher level resources such as Deployments or StatefulSets.
+
+When mounted as a volume, the contents of the Secret or Config Map will appear as files inside your Pod and its containers.
+Your application can just read and work with them as with any other files.
+The path where the files appear is configurable in the Pod specification, so you can mount them exactly where your application would expect them.
+You can also mount only selected files / fields instead of mounting the whole Secret or Config Map.
+When the contents of the Secret or Config Map changes, the files inside the Pod will be automatically updated (this does not happen immediately but in periodical intervals).
```
Comment (michalxo) [inline _posts/2021-07-22-using-kubernetes-config-provider-to-load-data-from-secrets-and-config-maps.md:24]: When content of Secret or Config Map changes, files inside the Pod will also be automatically updated. 
(feel free to ignore my suggestions, as I am not native speaker :) )

File: _posts/2021-07-22-using-kubernetes-config-provider-to-load-data-from-secrets-and-config-maps.md [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+* [Using Config Maps as environment variables](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#define-container-environment-variables-using-configmap-data)
+
+In many cases, these two mechanisms are fully sufficient.
+However, they have also some limitations:
+
+* They can be used only by applications running inside the Kubernetes cluster as Pods.
+  Applications running outside the Kubernetes cluster cannot use Secrets or Config Maps in this way
+* The Secrets and Config Maps you can load have to be in the same namespace as your application.
+  If they are in different namespace, you would need to copy them into your application namespace.
+* Any changes - such as adding an additional volume or environment variable - require a restart of the whole Pod.
```
Comment (michalxo) [inline _posts/2021-07-22-using-kubernetes-config-provider-to-load-data-from-secrets-and-config-maps.md:43]: Any change -

**Pull request merged**